### DISCUSSION TITLE
Convert jpg image to tiff with size less then 150KB

### DISCUSSION THREAD

**Discussion by sbind29893** [ImageMagick/ImageMagick#4006] [Help]
Hi team,

I need help in converting jpeg images to tiff images, final output tiff image size should be less then 150KB.

Thanks in advance.

**Comment by fmw42** [comment]
How big are you JPG files in width and height? Are they color or grayscale? What is your ImageMagick version and platform? What kind of TIFF compression do you need? Are you limited to any compress type?

**Reply by sbind29893** [reply]
hi @fmw42 ,

Thanks for your reply, pls find the below details.

JPG Files : Height & Width is not fixed.
Color or Grayscale : Both
Imagick Version : ImageMagick 6.9.10-23
Platform : Ubuntu 20.04
TIFF Compression : To be very honest i am not sure about any fixed compression technique. my only limitation is, there is this website where we need to upload tif images and it is not accepting images where size is greater then 150KB.

It would be great if you can help me.
Thanks

**Reply by fmw42** [reply]
You can only compress so much with TIFF (unless it is JPG compression). So the only way to reduce file size if compression does not work well enough is reducing the image dimensions. So the process depends upon how big in width and height and number of channels you have for the input. This process is one of iterating the dimensions until your file size is reached. I have a bash unix (imagemagick) shell script that does that. See my downsize script at http://www.fmwconcepts.com/imagemagick

**Reply by sbind29893** [reply]
Thanks for your reply, I am using your bash script on my linux system. I am getting below errors.

convert-im6.q16: invalid argument for option `-resize': -nan% @ error/convert.c/ConvertImageCommand/2592.

**Parameter used : -s 145**

**Reply by fmw42** [reply]
Post the image you are trying to use and I will check it out. It is possible things have changed in ImageMagick since I created it. Or it is possible that you have too much meta data (profiles, etc) and the script cannot reduce the file any smaller. Try doing a -strip on your input file and see if that helps.

**Reply by sbind29893** [reply]
Pls find the sample images.

![10DDEFD8-1055-4F62-B6A2-4993C2636CFD](https://user-images.githubusercontent.com/33458318/127714827-20429d77-53b7-48ee-b21c-17d47d4f1f1a.jpeg)

**Reply by sbind29893** [reply]
![AEB72221-A312-4483-A78A-952BBDE8A441](https://user-images.githubusercontent.com/33458318/127714916-d22f2dfb-18f9-473c-b518-8c4a94d730db.jpeg)

**Comment by fmw42** [comment]
I have fixed some issues and the script seems to work fine now on your 3 images. Download the script again and try once more.

**Reply by sbind29893** [reply]
hi @fmw42 ,

thanks for your efforts. I downloaded your new script, although it is somewhere decreasing the size. But not in the range I need it to be and I am getting this below error while running the script.
convert-im6.q16: invalid argument for option `-resize': -nan% @ error/convert.c/ConvertImageCommand/2592

Note : 

Original jpeg size : 1003 KB
New tiff size : 304 KB
Command : ./downsize.sh -s 150 sample_testimage.jpg new_output.tif

Still not coming is range below 150KB. I would be great if you can share the exact command which you have used on my sample images with the parameters.

**Reply by fmw42** [reply]
Post this image so that I can test with it. Did you download the new script?  What is the revised date at the top of the script?

**Reply by sbind29893** [reply]
![D6FDEAC6-AA2E-48D6-B0CC-33F125C517B1](https://user-images.githubusercontent.com/33458318/127729306-84337f11-f99e-4ad4-9fea-e98deb53bb97.jpeg)


revised 7/30/2021

**Reply by fmw42** [reply]
I have fixed that bug. So download the script and try again.

**Reply by sbind29893** [reply]
thanks for your help, now the resize error is resolved but the images are not coming under 150 KB.

Command i am using : downsize.sh -s 150 input.jpg output.tif
input jpg size : 5MB
output tif size : 391KB

pls help.

**Reply by fmw42** [reply]
Post an example image that fails. I tested your image above and get 64KB, which is less than 150KB.

```
downsize -s 150 127729306-84337f11-f99e-4ad4-9fea-e98deb53bb97.jpeg x.tif
i=1; size=167988B
i=2; size=122578B
i=3; size=64468B
final size = 64468B

identify x.tif
x.tif TIFF 286x518 286x518+0+0 8-bit sRGB 64468B 0.010u 0:00.000

```

**Reply by sbind29893** [reply]
hi @fmw42 ,

You can try image available on below link.

https://i.pinimg.com/originals/26/1b/00/261b00a940dc05c4ca002909525147f4.jpg

Original image size : 838KB
Output size : 187KB

**Reply by fmw42** [reply]
Download it again and try that version.
